Client/game_sa/CEntitySA.cpp

```diff
@@ -628,6 +650,36 @@ bool CEntitySA::SetBoneRotation(eBone boneId, float yaw, float pitch, float roll
     return false;
 }
 
+bool CEntitySA::SetBoneRotationQuat(eBone boneId, float x, float y, float z, float w)
+{
+    RpClump* clump = GetRpClump();
+    printf("SetBoneRotationQuat: %f %f %f %f\n", x, y, z, w);
+    if (clump)
+    {
+        // updating the bone frame orientation will also update its children
+        // This rotation is only applied when UpdateElementRpHAnim is called
+        CAnimBlendClumpDataSAInterface* clumpDataInterface = *pGame->GetClumpData(clump);
+        AnimBlendFrameData*             frameData = clumpDataInterface->GetFrameDataByNodeId(boneId);
+        printf("SetBoneRotationQuat 2: %f %f %f %f\n", x, y, z, w);
+        if (frameData)
+        {
+            RtQuat* boneOrientation = &frameData->m_pIFrame->orientation;
+            printf("SetBoneRotationQuat 3: %f %f %f %f\n", x, y, z, w);
+            boneOrientation->imag.x = x;
+            boneOrientation->imag.y = y;
+            boneOrientation->imag.z = z;
+            boneOrientation->real = w;
+            CEntitySAInterface* theInterface = GetInterface();
+            if (theInterface)
+            {
+                theInterface->bDontUpdateHierarchy = false;
+            }
+            return true;
+        }
+    }
+    return false;
```

Comment:
what's wrong here? it returns true if clump was found, false otherwise

Reply:
```cpp
if(!clump)
    return false;
// code
```

Reply:
> ```c++
> if(!clump)
>     return false;
> // code
> ```

waste of time for improving something what does not need improving. This method fit into one view of regular monitor and has low cognitive complexity so by force improving it is a waste of time. If you open any other random file you notice many important functions in shape of pyramide of doom and too long to comprehend

Reply:
The thing is, guidelines arent fresh. They've been here for quite some time now.
It was decided on discord that early returns should be used (not that drastically as I use sometimes but still 😅)

Reply:
Early return in this place would be redundant. I'm against it in such places.

Reply:
Absolutely not. This is not one liner. This can be shortened. If you're against this change, make a proper issue to mta docs. For now lets follow the guidelines. This change isnt as drastic as my previous changes so it will be fine

Reply:
> Absolutely not. This is not one liner. This can be shortened. If you're against this change, make a proper issue to mta docs. For now lets follow the guidelines. This change isnt as drastic as my previous changes so it will be fine

Guidelines tell the we have to overuse early return?

Reply:
That isnt overusing in this case

Reply:
> That isnt overusing in this case

It's only your opinion.

Reply:
Opinion or not, early returns are set in guidelines. This is **the** very example from the docs.

Reply:
I have to agree with @TracerDS here. Early return in this case simply improves code readability.